src/coreclr/tools/Common/Compiler/Win32Resources/ResourceData.cs

```diff
@@ -4,25 +4,46 @@
 using System;
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.Linq;
 using System.Reflection.Metadata;
 using System.Reflection.PortableExecutable;
 
+#if !HOST_MODEL
 using ILCompiler.DependencyAnalysis;
-using Internal.TypeSystem.Ecma;
+#endif
 
+#if HOST_MODEL
+namespace Microsoft.NET.HostModel.Win32Resources
+#else
 namespace ILCompiler.Win32Resources
+#endif
 {
     /// <summary>
     /// Resource abstraction to allow examination
     /// of a PE file that contains resources.
     /// </summary>
     public unsafe partial class ResourceData
     {
+        /// <summary>
+        /// Initialize a ResourceData instance from a PE file
+        /// </summary>
+        /// <param name="peFile"></param>
+        public ResourceData(PEReader peFile)
```

Comment:
Nit: technically we should ifdef these out for ilc/crossgen - to avoid size bloat. But given that those two tools ships as AOTed images it will be trimmed away. Just wanted to point out that putting dead code in is not 100% free.